Web/Views/Viia/Transactions.cshtml

```diff
@@ -1,27 +1,78 @@
+        const PAGING_TOKEN_STORAGE_KEY = "paging_token";
+        
+        $(document).ready(function() {
+            let transactions = @Html.Raw(Json.Serialize(@Model.Transactions));
+            let pagingToken = "@Model.PagingToken";
+            
+            localStorage.setItem(PAGING_TOKEN_STORAGE_KEY, pagingToken);
+            localStorage.setItem(TRANSACTION_STORAGE_KEY, JSON.stringify(transactions));
+            renderTransactions();
+        });
+        
+        function renderTransactions() {
+            let transactions = JSON.parse(localStorage.getItem(TRANSACTION_STORAGE_KEY));
+            transactions.forEach(renderSingleTransaction);
+            localStorage.setItem(TRANSACTION_STORAGE_KEY, "[]");
+        }
+        
+        function renderSingleTransaction(transaction) {
+            let amountSpan = $('<span></span>').text(transaction.transactionAmount.value);
+            if(transaction.transactionAmount.value >= 0) {
+                amountSpan.addClass('text-success');
+            } else {
+                amountSpan.addClass('text-danger');
+            }
+            let amountContainer = $(`<div class="p-2"></div>`).append(amountSpan).append(` ${transaction.transactionAmount.currency}`);
+            let balanceContainer = $(`<div class="p-2"></div>`).text(`${transaction.balance.value} ${transaction.balance.currency}`);
+            let amountAndBalanceContainer = $(`<div class="flex-column" style="margin-right: 1rem;"></div>`).append(amountContainer).append(balanceContainer);
+            
+            let txTextContainer = $(`<div class="p-2"></div>`).text(transaction.originalText);
+            let dateContainer = $(`<div class="p-2"></div>`).text(new Date(transaction.date).toISOString());
+            let textAndDateContainer = $(`<div class="flex-column flex-fill"></div>`).append(txTextContainer).append(dateContainer);
+            
+            let transactionContainer = $(`<div class="d-flex account-container justify-content-between"
+                   onclick="window.location.href = '/viia/accounts/${transaction.accountId}/transactions/${transaction.id}/details'">`)
+                   .append(textAndDateContainer)
+                   .append(amountAndBalanceContainer);
+            
+            $('#transaction-container').append(transactionContainer);
+        }
+        
+        function loadMore() {
+            let pagingToken = localStorage.getItem(PAGING_TOKEN_STORAGE_KEY);
+            $.ajax({
+                            url: `/viia/accounts/@Model.Transactions[0].AccountId/transactions/${pagingToken}`,
+                            type: 'GET',
+                            dataType: "json",
+                            success: function(data) {
+                                if(data.pagingToken === null || data.pagingToken === undefined || data.pagingToken === '') {
+                                    $('#load-more-btn').remove();
+                                }
```

Comment:
I'm not sure I get how the paging token is working, but will this break if I have two tabs open with the sample app looking in different accounts' transaction lists?

Reply:
Yep. Probably should make paging token stored as hidden html input or something

Reply:
Moved paging token to hidden input and moved transactions to be in memory, since adding them temporarily to localstorage is not needed